mars/learn/contrib/pytorch/sampler.py

```diff
@@ -0,0 +1,255 @@
+# Copyright 1999-2020 Alibaba Group Holding Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Iterator, Optional, Sized, Sequence
+import math
+
+try:
+    import torch
+    from torch.utils.data import Sampler
+except ImportError: # pragma: no cover
+    torch = None
+    Sampler = object
+
+from ....utils import require_not_none
+
+
+@require_not_none(torch)
+class SequentialSampler(Sampler):
+    r""""Samples elements sequentially, always in the same order.
+
+    Args:
+        data_source (Dataset): dataset to sample from
+    """
+    data_source: Sized
+
+    def __init__(self, data_source):
+        self.data_source = data_source
+
+    def __iter__(self) -> Iterator[int]:
+        return iter(range(len(self.data_source)))
```

Comment:
Is it possible to sample a batch data instead of 1 record each time?

Reply:
Batch related settings are under torch.utils.data.DataLoader. To avoid making a conflict with torch.utils.data.DataLoader, I didn't consider the batch setting in sampler.